pipes-tests.js

```diff
@@ -1,91 +1,107 @@
+  var myPipe = new Pipes("stage1", "stage2", "stage3");
 
-  test.equal(myPipe.pipelines.length === 1, true);
-  test.equal(myPipe.pipelines[0] === "pipeline1", true);
+  test.equal(myPipe._stages.length === 3, true);
+  test.equal(myPipe._stages[0] === "stage1", true);
+  test.equal(myPipe._stages[1] === "stage2", true);
+  test.equal(myPipe._stages[2] === "stage3", true);
 });
 
-Tinytest.add('Multi-pipeline Pipe instance created', function (test) {
-  var myPipe = new Pipes(["pipeline1","pipeline2"]);
+Tinytest.add('Multi-action pipe with actions at multiple-stages created', function (test) {
+  var myPipe = new Pipes(["stage1","stage2","stage3"]);
 
-  test.equal(myPipe.pipelines.length === 2, true);
-  test.equal(myPipe.pipelines[0] === "pipeline1", true);
-  test.equal(myPipe.pipelines[1] === "pipeline2", true);
-});
+  myPipe.on("default","stage1",function (options) {
+    return options;
+  });
 
-Tinytest.add('Single-action is added to Pipe pipeline', function (test) {
-  var myPipe = new Pipes("pipeline1");
+  myPipe.on("default","stage2",function (options) {
+    return options;
+  });
 
-  myPipe.on("pipeline1", function (options) {
+  myPipe.on("default","stage3",function (options) {
     return options;
   });
 
-  test.equal(myPipe.actions["pipeline1"].length === 1, true);
-  test.equal(typeof myPipe.actions["pipeline1"][0] === 'function', true);
+  test.equal(!! myPipe.actions["default"], true);
+  test.equal(!! myPipe.actions["default"]["stage1"], true);
+  test.equal(!! myPipe.actions["default"]["stage2"], true);
+  test.equal(!! myPipe.actions["default"]["stage3"], true);
+  test.equal(typeof myPipe.actions["default"]["stage1"][0] === 'function', true);
+  test.equal(typeof myPipe.actions["default"]["stage2"][0] === 'function', true);
+  test.equal(typeof myPipe.actions["default"]["stage3"][0] === 'function', true);
 });
 
-Tinytest.add('Multiple-actions are added to multiple pipelines', function (test) {
-  var myPipe = new Pipes(["pipeline1","pipeline2"]);
+Tinytest.add('Multi-action single-stage pipe executes and returns correct result', function (test) {
+  var myPipe = new Pipes("stage1");
 
-  myPipe.on("pipeline1", function (options) {
-    return options;
+  myPipe.on("default","stage1",function (options) {
+    return options+" had";
   });
 
-  myPipe.on("pipeline2", function (options) {
-    return options;
+  myPipe.on("default","stage1",function (options) {
+    return options+" a little";
+  });
+
+  myPipe.on("default","stage1",function (options) {
+    return options+" lamb";
   });
 
-  test.equal(myPipe.actions["pipeline1"].length === 1, true);
-  test.equal(myPipe.actions["pipeline2"].length === 1, true);
-  test.equal(typeof myPipe.actions["pipeline1"][0] === 'function', true);
-  test.equal(typeof myPipe.actions["pipeline2"][0] === 'function', true);
+  var result = myPipe.do("default","Mary");
+  test.equal(result === 'Mary had a little lamb', true);
```

Comment:
`test.equal(result, 'Mary had a little lamb')` will yield much more informative results in the case of a failed assertion.